app/scripts/controllers/HomeCtrl.js

```diff
@@ -1,8 +1,9 @@
 (function() {
-  function HomeCtrl(Room, Message, $uibModal) {
+  function HomeCtrl(Room, Message, $uibModal, $cookies) {
       var home = this;
       home.rooms = Room.all;
       home.currentRoom = null;
+      home.currentUser = $cookies.get('blocChatCurrentUser');
```

Comment:
By setting this here it is set only once, when the app loads. That's great if the user already has a cookie but if they don't, and they add one, this line never knows about it and `home.currentUser` will remain undefined.